### PR TITLE
fix: websocket test now passes + added tls support

### PR DESCRIPTION
**Pull request by anoushk1234**

### CODE REVIEW

File: rust/utils/src/task/websocket.rs [hunk 1]
```diff
@@ -116,8 +116,9 @@ impl WebSocket {
                     if let Ok(value) = serde_json::from_str::<Value>(&text) {
                         for (id, definition) in &self.subscriptions {
                             let json_path =
-                                json_parse_task(value.clone(), definition.filter.as_str())
-                                    .unwrap_or(Value::Null);
+                                json_parse_task(value.clone(), definition.filter.as_str());
+
+                            let json_path = json_path.unwrap_or(Value::Null);
                             if json_path.is_null() {
                                 return;
```
Comment (gallynaut) [inline rust/utils/src/task/websocket.rs:123]: this should be `continue;`

Same with Line 127, might be able to combine that if check

File: rust/utils/src/task/websocket.rs [hunk 2]
```diff
@@ -171,88 +173,93 @@ impl WebSocket {
...
+                        if json_value["symbol"].is_string() {
+                            is_subscribed = true;
+                            ws.send(tokio_tungstenite::tungstenite::Message::text("Subscribed"))
+                                .await
+                                .unwrap();
+                        }
+                    }
+                }
+
+                if is_subscribed {
```
Comment (gallynaut) [inline rust/utils/src/task/websocket.rs:223]: if this is too big of an ask just ignore (alot of code just to test this)

We should have the test server store a map <symbol, destination> then allow the test client to subscribe to the first symbol, get a response, subscribe to a new symbol, test we are still getting a response from both symbols, then remove one of the subscriptions and test we still have some data coming through.

tl;dr have websocket support multiple subscriptions to the same host

File: rust/utils/src/task/websocket.rs [hunk 3]
```diff
@@ -134,6 +135,7 @@ impl WebSocket {
                                     data: value.clone(),
                                 },
                             );
+
                             return;
```
[the remark this answers is not in the document]
Comment (anoushk1234) [reply rust/utils/src/task/websocket.rs:139]: this makes the test run forever though
